terragrunt/modules/crates-io/cloudfront-functions/static-router.js

```diff
@@ -5,6 +5,9 @@ function handler(event) {
     // See more: https://github.com/rust-lang/crates.io/issues/4891
     if (request.uri.includes("+")) {
         request.uri = request.uri.replace("+", "%2B");
+    } else if (request.uri === '/archive/version-downloads/') {
```

Comment:
is this guaranteed to always end with a slash?

Reply:
Depends on the user and what they type into the address bar. 😅

I think the question is more: Do we want to support the URL without a trailing slash as well? Personally, I'd say No until someone complains. While it's not _that_ more work to change this into a regex, a simple comparison is easier to reason about and anything else might be premature optimization...

Reply:
I'd prefer to support both. It's too confusing otherwise and the point of the index file is discoverability.

Reply:
FWIW I'd also be fine with a redirect from no-slash to slash :)

Reply:
I'd prefer to redirect to include the slash so we can use relative links in the generated HTML, if that's not too hard.

Reply:
Isn't it what the code does at the moment?
Or am I missing something?

Reply:
> Isn't it what the code does at the moment?

it currently serves the HTML file at both locations.

- at `/archive/version-downloads/` a link like `<a href="2024-08-20.csv">` would lead to `/archive/version-downloads/2024-08-20.csv`
- at `/archive/version-downloads` a link like `<a href="2024-08-20.csv">` would lead to `/archive/2024-08-20.csv`

that's why Adam prefers an HTTP redirect ([301 status code + `Location` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301#server_response)) for the non-slash location, instead of serving the HTML there directly.

Reply:
Makes sense.
Ready for another look 👍